lib/ide/src/controller/text.rs

```diff
@@ -6,34 +6,30 @@
 
 use crate::prelude::*;
 use crate::controller::FallibleResult;
+use crate::controller::notification;
 
-use data::text::TextChangedNotification;
-use failure::_core::fmt::Formatter;
-use failure::_core::fmt::Error;
+use data::text::TextChangedInfo;
 use file_manager_client as fmc;
 use flo_stream::MessagePublisher;
-use flo_stream::Publisher;
 use flo_stream::Subscriber;
 use json_rpc::error::RpcError;
 use shapely::shared;
+use crate::executor::global::spawn;
 
 
-// ====================
-// === Notification ===
-// ====================
 
-/// A buffer size for notification publisher.
-///
-/// If Publisher buffer will be full, the thread sending next notification will be blocked until
-/// all subscribers read message from buffer. We don't expect much traffic on file notifications,
-/// therefore there is no need for setting big buffers.
-const NOTIFICATION_BUFFER_SIZE : usize = 36;
-
-/// A notification from TextController.
-#[derive(Clone,Debug)]
-pub enum Notification {
-    /// File contents needs to be set to the following due to synchronization with external state.
-    SetNewContent(String),
+async fn process_stream_with_handle<Stream,Weak,F>(mut stream:Stream, weak:Weak, mut function:F)
```

Comment:
First I thought it should be implemented using StreamExt::map, but now I'm not convinced.
Just give it a fleeting thought. Likely it is not a good idea.

Reply:
Not, because I want to stop reading stream once we know that handle is no more. With `map` function I cannot say "stop reading this stream anymore". I could use something like "filter_map", but still this is stinky.

Reply:
true